clang/include/indexstore/indexstore.h

```diff
@@ -287,7 +291,7 @@ typedef enum {
   INDEXSTORE_SYMBOL_SUBKIND_SWIFTACCESSORMODIFY = 1015,
 } indexstore_symbol_subkind_t;
 
-typedef INDEXSTORE_OPTIONS(uint64_t, indexstore_symbol_property_t) {
+INDEXSTORE_OPTIONS(uint64_t, indexstore_symbol_property_t) {
```

Comment:
When `__has_feature(objc_fixed_enum)` is false,  `INDEXSTORE_OPTIONS ` gets defined as

```c
#ifndef INDEXSTORE_OPTIONS
# define INDEXSTORE_OPTIONS(_type, _name) _type _name; enum INDEXSTORE_OPTIONS_ATTRS
#endif
```

By removing the `typedef` at the macro callsites, the macro expansions would change, for example it currently expands to:

```c
typedef uint64_t indexstore_symbol_property_t; enum [...] {
```

with this change it expands to:

```c
uint64_t indexstore_symbol_property_t; enum [...] {
```

In my quick test, I believe this should result in an error.

```
input.c:1:19: error: redefinition of 'x' with a different type: 'enum
      (anonymous enum at input.c:1:8)' vs 'int'
```

Reply:
resolved